perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1246,30 +1248,49 @@ public void removeAllPreferredEmailChangeRequests(PerunSession sess, User user)
 	}
 
 	@Override
-	public Pair<String,String> loadPasswordResetRequest(User user, int requestId) {
-
-		int validWindow = BeansUtils.getCoreConfig().getPwdresetValidationWindow();
+	public void checkPasswordResetRequestIsValid(PerunSession sess, User user, int requestId) throws PasswordResetLinkExpiredException, PasswordResetLinkNotValidException {
+		this.checkAndGetPasswordResetRequest(user, requestId);
+	}
 
+	private Pair<String, String> checkAndGetPasswordResetRequest(User user, int requestId) throws PasswordResetLinkExpiredException, PasswordResetLinkNotValidException {
 		try {
+			int numberOfRequests = jdbc.queryForInt("select count(1) from pwdreset where user_id=? and id=?", user.getId(), requestId);
+			if (numberOfRequests == 0) {
+				throw new PasswordResetLinkNotValidException("Password request " + requestId + " doesn't exist for user: " + user);
```

Comment:
I would omit ``user:`` from exception message, since ``user.toString()`` prints also object name. Same for the exception thrown below and at the end.